cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java

```diff
@@ -172,6 +180,83 @@ public static CassandraNodeTask getTaskForNode(@NotNull final CassandraNode cass
         return builder;
     }
 
+    @NotNull
+    public static Function<Resource, TreeSet<Long>> resourceToPortSet() {
+        return new Function<Resource, TreeSet<Long>>() {
+            @Override
+            @NotNull
+            public TreeSet<Long> apply(@Nullable Resource resource) {
+                return resourceValueRange(Optional.fromNullable(resource));
+            }
+        };
+    }
+
+    public static Predicate<Resource> containsPorts(final Iterable<Long> ports) {
+        return new Predicate<Resource>() {
+            @Override
+            public boolean apply(Resource resource) {
+                TreeSet<Long> portsInResource = resourceValueRange(Optional.fromNullable(resource));
+                return portsInResource.containsAll(newArrayList(ports));
+            }
+        };
+    }
+
+    public static ImmutableListMultimap<String, Resource> resourcesForRoleAndOffer(String role, Protos.Offer offer) {
+        return from(offer.getResourcesList())
+                .filter(resourceHasExpectedRole(role))
+                .index(resourceToName());
+    }
+
+    public static Predicate<Resource> scalarValueAtLeast(final long v) {
+        return new Predicate<Resource>() {
```

Comment:
Please make a private static final inner class of this predicate with a single constructor argument of the passed in `v`.

See https://github.com/mesosphere/cassandra-mesos/blob/master/cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java#L236-L248 for an example of how this had been done previously.